.github/workflows/ci.yml

```diff
@@ -0,0 +1,368 @@
+name: ci
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+env:
+  GRADLE_OPTS: "-Xmx4g -Dorg.gradle.daemon=false"
+  total-runners: 8
+
+jobs:
+  compile:
+    runs-on: ubuntu-latest
```

Comment:
I'm wary of latest, as it may undergo a distro update underneath us at some point.

latest is `ubuntu-20.04`, but perhaps `ubuntu-22.04`?  we may need 22.04 for java 17 support.

This comment also is for _all_ instances of ubuntu-latest in the PR.  I don't want to litter it with comments.

Reply:
ok, resolved.